### ISSUE TITLE
🐞 [Bug]: method that requires state, but does not switch_to, returns stateless object.

### ISSUE THREAD

**Issue by jkneer** [ozgunozerk/state-shift#36]
### What happened?

```rust
#[type_state(
    states = (Unset, FullyConfigured),
    slots = (Unset)
)]
struct BobTheBuilder {}

#[impl_state]
impl BobTheBuilder {
  #[require(Unset)]
  #[switch_to(FullyConfigured)]
  pub fn init() -> BobTheBuilder {
    BobTheBuilder {}
  }
}

  #[require(FullyConfigured)]
  pub fn dosth(self, Option<u32>) -> BobTheBuilder {
    BobTheBuilder {}
  }

```

In this case the call to `dosth()` will return a stateless BobTheBuilder.
This is unexpected, because this basically switches the state from `FullyConfigured` to stateless without a switch_to statement. On the other hand this can be fixed by adding a switch_to macro to the initial state.
So this implementation

```rust
  #[require(FullyConfigured)]
  #[switch_to(FullyConfigured)]
  pub fn dosth(self, Option<u32>) -> BobTheBuilder {
    BobTheBuilder {}
  }
```
works as expected.

### Expected behavior

This behaviour took me by surprise. I would have expected that the OutputState = InputState as long as I do not call switch_to and that I would not need to add switch_to as long as I just want to require a state and update the struct, but do not want to change the state. The current behaviour is kind of the inverse of what I expected by intuition.

**Comment by ozgunozerk** [comment]
You are absolutely right, if `switch_to` is not specified, it should not preserve the type. 

Will deal with this soon, thanks for the report 💯

**Comment by ozgunozerk** [comment]
This one is resolved, will issue a new release after #35 is also resolved
